UI/auth-youtube.cpp

```diff
@@ -0,0 +1,223 @@
+#include "auth-youtube.hpp"
+
+#include <iostream>
+#include <QMessageBox>
+#include <QThread>
+#include <vector>
+#include <QHBoxLayout>
+
+#ifdef WIN32
+#include <windows.h>
+#include <shellapi.h>
+
+#pragma comment(lib, "shell32")
+#endif
+
+#include "auth-listener.hpp"
+#include "obs-app.hpp"
+#include "qt-wrappers.hpp"
+#include "ui-config.h"
+#include "youtube-api-wrappers.hpp"
+#include "window-basic-main.hpp"
+#include "obf.h"
+
+using namespace json11;
+
+/* ------------------------------------------------------------------------- */
+#define YOUTUBE_AUTH_URL "https://accounts.google.com/o/oauth2/v2/auth"
+#define YOUTUBE_TOKEN_URL "https://www.googleapis.com/oauth2/v4/token"
+#define YOUTUBE_SCOPE_VERSION 1
+#define YOUTUBE_API_STATE_LENGTH 32
+#define SECTION_NAME "YouTube"
+
+static const char allowedChars[] =
+	"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+static const int allowedCount = static_cast<int>(sizeof(allowedChars) - 1);
+/* ------------------------------------------------------------------------- */
+
+const QString ReplaceAll(QString str, const QChar ch, const QString &to)
+{
+	int start_pos = 0;
+	while ((start_pos = str.indexOf(ch, start_pos)) != std::string::npos) {
+		str.replace(start_pos, 1, to);
+		start_pos += to.length();
+	}
+	return str;
+};
+
```

Comment:
Remove this as it's no longer necessary as QUrl and QDesktopServices do all the heavy lifting.

```suggestion
```